system/applications/modules/mullvad.nix

```diff
@@ -0,0 +1,17 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib) mkIf optional;
+
+  cfg = config.icedos.applications;
+in
+mkIf (cfg.mullvad.enable) {
+  environment.systemPackages = [ ] ++ optional (cfg.mullvad.gui) pkgs.mullvad-vpn;
```

Comment:
There's no need for optional here. Just use mkIf. 
```nix
environment.systemPackages = mkIf (cfg.mullvad.gui) [ pkgs.mullvad-vpn ];
```

Reply:
Yep, was testing something else and forgot to change it back, fixed.